validphys2/src/validphys/theorycovariance/construction.py

```diff
@@ -182,10 +183,10 @@ def dataset_names(data_input):
     return [el.name for el in data_input]
 
 
-ProcessInfo = namedtuple("ProcessInfo", ("theory", "namelist", "sizes"))
+ProcessInfo = namedtuple("ProcessInfo", ("theory", "namelist", "sizes", "data"))
 
 
-def combine_by_type(each_dataset_results_bytheory, dataset_names):
+def combine_by_type(each_dataset_results_bytheory, dataset_names, groups_dataset_inputs_loaded_cd_with_cuts):
```

Comment:
I don't think this would work since `each_dataset_results_bytheory` collects over `theorids`. If you'd like to use this `combine_by_type` function you could consider replacing `each_dataset_results_bytheory` by a `produce_` rule. You could do this in such a way where if the thcovmat being calculated is the usual MHOU it loops over different theoryids, but if the theorycovmat is a HT comvat the predictions corresponding to the up and down shift are calculated from a single predicion but provided in the same format as the `each_dataset_results_bytheory`.